### DISCUSSION TITLE
Do we allow to use covariates for data in prediction period when doing prediction

### DISCUSSION THREAD

**Discussion by yuhao-morning** [PaddlePaddle/PaddleSpatial#53] [BaiduKDDCup2022] [answered]
Instead of using several days data before the prediction period, can we use covariates of data in prediction period to predict the targets in this period?

**Comment by zhoujingbo** [comment]
Hi,

What does the mean of covariates of data? 

In general, we only limit the input of the model, and we do not give any restrictions on how to build the model. 

Thanks
Regards

**Comment by yuhao-morning** [comment]
Also, can you answer this question?


For example, we use 7days to predict 48 hours Patv, then our model's input is the data of those 7 days,  and the outputs the Patv of next 48 hours. But another possibility is: our model's input is the data of those 7 days PLUS the features of next 48 hours data, here the features mean, like, "TurbID",
        "Day",
        "Tmstamp",
        "Wspd",
        "Wdir",
        "Etmp",
        "Itmp",
        "Ndir",
        "Pab1",
        "Pab2",
        "Pab3",
        "Prtv",
        "x",
        "y",
        "hour",
                            "minute",

and we use these features to help to predict the target Patv of these 48 hours.  Is the second one allowed ?






At 2022-05-17 12:15:21, "zhou jingbo" ***@***.***> wrote:
Hi,
What does the mean of covariates of data?
In general, we only limit the input of the model, and we do not give any restrictions on how to build the model.
Thanks
Regards

**Comment by ramber1836** [comment, marked as answer]
Sorry, you cannot use the future features as input to predict next 48 hours
Patv.

Regards,
Xinjiang Lu

**Comment by yuhao-morning** [comment]
For example, we use 7days to predict 48 hours Patv, then our model's input is the data of those 7 days,  and the outputs the Patv of next 48 hours. But another possibility is: our model's input is the data of those 7 days PLUS the features of next 48 hours data, here the features mean, like, "TurbID",

        "Day",

        "Tmstamp",

        "Wspd",

        "Wdir",

        "Etmp",

        "Itmp",

        "Ndir",

        "Pab1",

        "Pab2",

        "Pab3",

        "Prtv",

        "x",

        "y",

        "hour",

                            "minute",


and we use these features to help to predict the target Patv of these 48 hours.  Is the second one allowed ?

















At 2022-05-17 12:15:21, "zhou jingbo" ***@***.***> wrote:

Hi,

What does the mean of covariates of data?

In general, we only limit the input of the model, and we do not give any restrictions on how to build the model.

Thanks
Regards
